src/Uno.Extensions.Reactive/Collections/Tracking/CollectionAnalyzer.Core.cs

```diff
@@ -0,0 +1,229 @@
+		RichNotifyCollectionChangedEventArgs arg,
+		Func<IList, ListRef<T>> getRef,
+		ComparerRef<T>? versionComparer)
+	{
+		switch (arg.Action)
+		{
+			case NotifyCollectionChangedAction.Add:
+			case NotifyCollectionChangedAction.Remove:
+			case NotifyCollectionChangedAction.Move:
+			case NotifyCollectionChangedAction.Reset:
+				return new _Event<T>(arg);
+
+			// For the replace, we are digging in the collection to determine if we could simplify the change considering the given comparers
+			case NotifyCollectionChangedAction.Replace:
+				return GetChangesCore(getRef(arg.OldItems), getRef(arg.NewItems), versionComparer, arg.OldStartingIndex);
+
+			default:
+				throw new ArgumentOutOfRangeException(nameof(arg), arg.Action, $"Action '{arg.Action}' not supported.");
+		}
+	}
+
+	internal static Change<T>? GetChangesCore<T>(
+		ListRef<T> oldItems,
+		ListRef<T> newItems,
+		ComparerRef<T>? versionComparer,
+		int eventArgsOffset = 0)
+	{
+		//if (oldItems.Instance is IDifferentialCollection oldDiff
+		//	&& newItems.Instance is IDifferentialCollection newDiff
+		//	&& oldDiff.FindCommonAncestor(newDiff) is { } ancestor
+		//	&& ancestor == oldDiff.Head) // Temp
+		//{
+		//	// 1. Rollback changes from oldDiff to ancestor
+		//	// => Not supported yet, see condition 'ancestor == oldDiff.Head' above
+
+		//	// 2. Apply changes from ancestor to newDiff
+		//	new 
+		//}
+
+		/*
+		* OLD: the source collection we are going to update to the NEW
+		* NEW: the collection that we want to go to
+		* RESULT (a.k.a. virtual old) : This makes sense only while synchronizing, it's represents the OLD with the changes already applied. 
+		*								 At the end it's expected to be sequence equals to the NEW.
+		* 
+		* We detect changes to update from OLD to NEW. We could have choose the other way, but as the consumer (i.e. the view) knows the OLD, it's a bit more logical.
+		* 
+		* Note: comparer may be null for IObservableCollectionSnapshot.IndexOf(). 
+		*		 It's more performant to forward the 'null' instead of defaulting to EqualityComparer.Default
+		* 
+		*/
+
+		int added = 0, moved = 0, removed = 0;
+		var buffer = new ChangesBuffer<T>(oldItems.Count, newItems.Count, eventArgsOffset);
+
+		var oldEnumerator = new SourceEnumerator<T>(oldItems);
+		while (oldEnumerator.MoveNext())
+		{
+			var oldIndex = oldEnumerator.CurrentIndex;
+			var oldItem = oldEnumerator.Current!;
+			var resultIndex = oldIndex - removed + added + moved - oldEnumerator.Ignored; // The current index in the (virtual) result collection (i.e. oldIndex ignoring the removed/added items)
+			var newIndex = newItems.IndexOf(oldItem, resultIndex, newItems.Count - resultIndex);
+
+			if (newIndex < 0)
+			{
+				// Item is no more present in the new collection : Remove it
+
+				buffer.Remove(oldItem, resultIndex);
+				removed++;
+
+				continue;
+			}
+
+			if (newIndex < resultIndex)
+			{
+				throw new InvalidOperationException("The index return by the IndexOf is invalid");
```

Comment:
Why can this happen ? Can it be actionable?

```suggestion
				throw new InvalidOperationException("The index returned by the IndexOf is invalid");
```